src/wazuh_modules/agent_upgrade/wm_agent_upgrade.h

```diff
@@ -103,4 +108,21 @@ cJSON *wm_agent_upgrade_process_upgrade_custom_command(const cJSON* params, cons
  * */
 cJSON* wm_agent_upgrade_process_upgrade_result_command(const cJSON* agents);
 
+/**
+ * Check if agent exist
+ * @param agent_id Id of agent to validate
+ * @return error_code (0 = success, -1 = agent not exist)
+ * */
+int wm_agent_upgrade_validate_id(int agent_id);
+
+/**
+ * Check if agent version is valid to upgrade
+ * @param agent_id Id of agent to validate
+ * @return error_code (0 = not error,   WM_UPGRADE_NOT_MINIMAL_VERSION_SUPPORTED, 
+ *                                      WM_UPGRADE_VERSION_SAME_MANAGER, 
+ *                                      WM_UPGRADE_NEW_VERSION_LEES_OR_EQUAL_THAT_CURRENT,
+ *                                      WM_UPGRADE_NEW_VERSION_GREATER_MASTER)
```

Comment:
The different options for return should be specified using **retval**
```
@return error_code
@retval 0 not error
@retval WM_UPGRADE_NOT_MINIMAL_VERSION_SUPPORTED
....
```

Reply:
you're right, solved